[CRASH] R8 stripping `net.thunderbird.core.common.mail.Flag`, used by Moshi

Both `@ceye4n:matrix.org` and `@baardfigur:matrix.org` reported experiencing a crash in the daily build when attempting to add a new account.

The recent migration to AGP9 required us to switch from using `proguard-android.txt` to `proguard-android-optimize.txt`, which has led to some unpredictable side effects.

Currently, because the release builds are running with optimization, R8 is stripping away certain classes that are accessed via reflection, such as those used on serialization with Moshi.

Stacktrace:
```
FATAL EXCEPTION: main (Show original) (Fix with AI)
Process: net.thunderbird.android.daily, PID: 8657
java.lang.AssertionError: Missing field in net.thunderbird.core.common.mail.Flag
	at com.squareup.moshi.StandardJsonAdapters$EnumJsonAdapter.<init>(StandardJsonAdapters.java:282)
	at com.squareup.moshi.StandardJsonAdapters$1.create(StandardJsonAdapters.java:68)
	at com.squareup.moshi.Moshi.adapter(Moshi.java:146)
	at com.squareup.moshi.ClassJsonAdapter$1.createFieldBindings(ClassJsonAdapter.java:145)
	at com.squareup.moshi.ClassJsonAdapter$1.create(ClassJsonAdapter.java:107)
	at com.squareup.moshi.Moshi.adapter(Moshi.java:146)
	at com.squareup.moshi.Moshi.adapter(Moshi.java:106)
	at com.squareup.moshi.Moshi.adapter(Moshi.java:80)
	at com.fsck.k9.controller.PendingCommandSerializer.<init>(PendingCommandSerializer.java:45)
	at com.fsck.k9.controller.PendingCommandSerializer.<clinit>(PendingCommandSerializer.java:26)
	at com.fsck.k9.controller.PendingCommandSerializer.getInstance(PendingCommandSerializer.java:53)
	at com.fsck.k9.mailstore.LocalStore.<init>(LocalStore.java:180)
	at com.fsck.k9.mailstore.LocalStore.createInstance(LocalStore.java:172)
	at com.fsck.k9.mailstore.LocalStoreProvider.getInstance$lambda$0(LocalStoreProvider.kt:23)
	at com.fsck.k9.mailstore.LocalStoreProvider.getInstanceById(LocalStoreProvider.kt:46)
	at com.fsck.k9.mailstore.LocalStoreProvider.getInstance(LocalStoreProvider.kt:22)
	at com.fsck.k9.mailstore.SpecialLocalFoldersCreator.createSpecialLocalFolders(SpecialLocalFoldersCreator.kt:21)
	at net.thunderbird.app.common.account.AccountCreator.create(AccountCreator.kt:87)
	at net.thunderbird.app.common.account.AccountCreator.access$create(AccountCreator.kt:33)
	at net.thunderbird.app.common.account.AccountCreator$create$1.invokeSuspend(AccountCreator.kt)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:34)
                                                                                                    	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:98)
                                                                                                    	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.java:124)
                                                                                                    	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:89)
                                                                                                    	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:586)
                                                                                                    	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:820)
                                                                                                    	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:717)
                                                                                                    	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:704)
	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@a1d137b, Dispatchers.Main.immediate]
Caused by: java.lang.NoSuchFieldException: RECENT
	at java.lang.Class.getField(Class.java:2327)
	at com.squareup.moshi.StandardJsonAdapters$EnumJsonAdapter.<init>(StandardJsonAdapters.java:278)
	... 19 more
```

--- a/app-k9mail/proguard-rules.pro
+++ b/app-k9mail/proguard-rules.pro
@@ -62,3 +62,13 @@
 -dontwarn org.slf4j.impl.StaticLoggerBinder
 
 -keep,allowshrinking class com.tokenautocomplete.TokenCompleteTextView
+
+# Moshi's EnumJsonAdapter uses Class.getField() to map JSON strings to enum constants.
+# R8's enum optimization rewrites switch statements to use ordinals, making enum fields
+# appear unused, which causes R8 to strip them. This rule preserves all enum constants
+# so Moshi can deserialize them at runtime (e.g. via PendingCommandSerializer).
+-keep class net.thunderbird.core.common.mail.Flag { *; }
+
+# Moshi uses reflection to read field types. R8 can rewrite List/Map to ArrayList/HashMap,
+# which breaks Moshi's collection interface requirement.
+-keep class com.fsck.k9.controller.MessagingControllerCommands$* { *; }

--- a/app-thunderbird/proguard-rules.pro
+++ b/app-thunderbird/proguard-rules.pro
@@ -62,3 +62,13 @@
 -dontwarn org.slf4j.impl.StaticLoggerBinder
 
 -keep,allowshrinking class com.tokenautocomplete.TokenCompleteTextView
+
+# Moshi's EnumJsonAdapter uses Class.getField() to map JSON strings to enum constants.
+# R8's enum optimization rewrites switch statements to use ordinals, making enum fields
+# appear unused, which causes R8 to strip them. This rule preserves all enum constants
+# so Moshi can deserialize them at runtime (e.g. via PendingCommandSerializer).
+-keep class net.thunderbird.core.common.mail.Flag { *; }
+
+# Moshi uses reflection to read field types. R8 can rewrite List/Map to ArrayList/HashMap,
+# which breaks Moshi's collection interface requirement.
+-keep class com.fsck.k9.controller.MessagingControllerCommands$* { *; }

